driver.py

```diff
@@ -1,1282 +1,29 @@
-
-
-    def inference_colocated(self,
-                            exp_name="inference-scaling",
-                            launcher="auto",
-                            nodes=[12],
-                            clients_per_node=[18],
-                            db_cpus=[2],
-                            db_tpq=[1],
-                            db_port=6780,
-                            pin_app_cpus=[False],
-                            batch_size=[1],
-                            device="GPU",
-                            num_devices=1,
-                            net_ifname="ipogif0",
-                            rebuild_model=False
-                            ):
-        """Run ResNet50 inference tests with colocated Orchestrator deployment
-
-        :param exp_name: name of output dir, defaults to "inference-scaling"
-        :type exp_name: str, optional
-        :param launcher: workload manager i.e. "slurm", "pbs"
-        :type launcher: str, optional
-        :param nodes: compute nodes to use for synthetic scaling app with
-                      a co-located orchestrator database
-        :type clients_per_node: list, optional
-        :param clients_per_node: client tasks per compute node for the synthetic scaling app
-        :type clients_per_node: list, optional
-        :param db_hosts: optionally supply hosts to launch the database on
-        :type db_hosts: list, optional
-        :param db_nodes: number of compute hosts to use for the database
-        :type db_nodes: list, optional
-        :param db_cpus: number of cpus per compute host for the database
-        :type db_cpus: list, optional
-        :param db_tpq: number of device threads to use for the database
-        :type db_tpq: list, optional
-        :param db_port: port to use for the database
-        :type db_port: int, optional
-        :param pin_app_cpus: pin the threads of the application to 0-(n-db_cpus)
-        :type pin_app_cpus: int, optional
-        :param batch_size: batch size to set Resnet50 model with
-        :type batch_size: list, optional
-        :param device: device used to run the models in the database
-        :type device: str, optional
-        :param num_devices: number of devices per compute node to use to run ResNet
-        :type num_devices: int, optional
-        :param net_ifname: network interface to use i.e. "ib0" for infiniband or
-                           "ipogif0" aries networks
-        :type net_ifname: str, optional
-        :param rebuild_model: force rebuild of PyTorch model even if it is available
-        :type rebuild_model: bool
-        """
-        logger.info("Starting colocated inference scaling tests")
-        logger.info(f"Running with database backend: {_get_db_backend()}")
-
-        self._check_model(device, force_rebuild=rebuild_model)
+class SmartSimScalingTests(Inference, Throughput, DataAggregation):
+    def driver():
```

Comment:
Im not sure why I made it like that now that Im looking at it. The only case someone would call python driver.py driver would be due to a mistake lol. 

Undefined symbols? What is the difference adding self. ? I will most definitely change it, is it more of a python style thing?

Reply:
also! to get the tests that use start_db to work, you have to pass in --node_feature=P100, I should set a default for that, Ill do that

Reply:
> Undefined symbols? What is the difference adding self. ? I will most definitely change it, is it more of a python style thing?

Oh! python doesn't automatically look for fns defined on the "this/self" pointer like C++/Java. It will automatically look for unbound fns defined at the current scope up to the top level scope if `self.` is not used.

So in this case:

```py
class C:
    def my_func(self):  # <-- despite being defined here
        ...

    def run(self):
        my_func()  # <-- this will raise a ``NameError``

# Because python will skip over the class definition
# and look for the function at the top level for an unbound function


# The "correct" way to call the method in question would be to call
# the "bound" func through `self`
        self.my_func()

Reply:
I was starring at your example for such a long time, I remember what you are referring to now, thank u!